etc/cmake/options.cmake

```diff
@@ -88,7 +88,24 @@ if (OTBR_SRP_ADVERTISING_PROXY)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_SRP_ADVERTISING_PROXY=1)
 endif()
 
+option(OTBR_USE_OT_SRP_ADV_PROXY "Enable OT core Advertising Proxy" OFF)
+
+if (OTBR_USE_OT_SRP_ADV_PROXY AND OTBR_SRP_ADVERTISING_PROXY)
+    message(FATAL_ERROR "Only one Advertising Proxy can be enabled.")
+endif()
+
+option(OTBR_SRP_SERVER "Enable SRP server" OFF)
+if (OTBR_USE_OT_SRP_ADV_PROXY OR OTBR_SRP_ADVERTISING_PROXY)
+    set(OTBR_SRP_SERVER ON CACHE STRING "Enable SRP server" FORCE)
```

Comment:
I think 'OTBR_SRP_SERVER' here is a 'derived value' so we probably should not provide a way to configure it. It should be only enabled when ADVERTISTING_PROXY (either OT impl or ot-br-posix impl) is enabled. So what about we calculate this derived value in a header in ot-br-posix and use the value in sources by including that header.

As I checked, it seems there is no appropriate header for now. I suggest we add a new header called `derived-config.h`. I believe this can also be used in other scenarios.